pkg/flowkit/services/transactions.go

```diff
@@ -80,6 +77,56 @@ func (t *Transactions) GetStatus(
 	return tx, result, err
 }
 
+// BuildWithSigners builds a transaction with a set of signers.
+func (t *Transactions) BuildWithSigners(
+	signers []*flowkit.Account,
+	code []byte,
+	codeFilename string,
+	gasLimit uint64,
+	args []cadence.Value,
+	network string,
+	approveBuild bool,
+) (*flowkit.Transaction, error) {
+
+	//find authorizer count from code
+	authorizerCount := flowkit.GetAuthorizerCount(codeFilename, code)
+
+	var seen map[flow.Address]any = make(map[flow.Address]any)
+	var authorizers []flow.Address
+	for _, auth := range signers {
+		if _, ok := seen[auth.Address()]; ok {
+			continue
+		}
+		authorizers = append(authorizers, auth.Address())
+	}
+
+	if len(authorizers) < authorizerCount {
+		return nil, fmt.Errorf("invalid number of authorizers, expected: %d", authorizerCount)
+	}
```

Comment:
wouldn't that be handled by https://github.com/bluesign/flow-cli/blob/c820e7b9125d75b12e85b28afbd9b796461d7a09/pkg/flowkit/transaction.go#L317

Reply:
yeah, I will move the check to the slice one line below.